### ISSUE TITLE
curly rule improment 

### ISSUE THREAD

**Issue by GodofHope** [palantir/tslint#4822]
I suggest that the curly rule should not show warning or errors if a if branch contains anothor ifelse branch.

Here are my code, obviously I cannot remove the branket, but tslint give me warnings while eslint not:  
```js
  for (let i = 0; i < length; i += 1)
     if (A) {
       if (B ) {
         do somthing M...
         break;
       } else if (D)
           do somthing N...
     } else do somthing O...
```

Hope you can improve it.

Errors:
```
Block contains only one statement; remove the curly braces. (curly)
Block contains only one statement; remove the curly braces. (curly)
Block contains only one statement; remove the curly braces. (curly)
```

**Comment by JoshuaKGoldberg** [comment]
Hi @GodofHope! Although your issue is a valid use case, *TSLint is being deprecated* and no longer accepting pull requests for formatting changes such as this one. See #4534.

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
